lib/Connection.js

```diff
@@ -123,10 +124,28 @@ class Connection {
       callback(new RequestAbortedError(), null)
     }
 
+    const onSocket = socket => {
+      /* istanbul ignore else */
+      if (!socket.isSessionReused()) {
+        socket.once('secureConnect', () => {
+          // Check if fingerprint matches
+          /* istanbul ignore else */
+          if (this.caFingerprint !== getIssuerCertificate(socket).fingerprint256) {
+            onError(new Error('Fingerprint does not match'))
```

Comment:
```suggestion
            onError(new Error('Server certificate CA fingerprint does not match the value configured in  caFingerprint'))
```

Not sure how detailed we want to be in our error messages in the client but I found the above to be a bit cryptic